src/cli.ts

```diff
@@ -94,18 +93,51 @@ if(opts["headless"]) {
     program.outputHelp();
     console.error("\nERROR: Unable to run Eve headless unless you specify a program to run with `<file>`");
     process.exit(1);
-  }
+  } 
+  
+  let filepath = config.file!.replace("file", config.workspacePaths["file"]);
 
+  if(!fs.existsSync(filepath)) {
+    console.error(`\nERROR: File "${filepath}" was not found.`);
+    process.exit(1);
+  }
+  
   console.info(`Starting headless Eve instance...`);
   console.info("  Requiring watchers from include paths...");
 
   for(let watcherFile of findWatchers(config.watcherPaths)) {
     require(watcherFile);
   }
 
-  console.info("  Starting program...");
-
-  require(config.file!.replace("file", config.workspacePaths["file"]));
+  console.info(`  Starting program ${filepath}...`);
+
+  let extension = filepath.split('.').pop();
+  if(extension === "eve") {
+    console.info("running an eve file");
+    fs.readFile(filepath, {encoding: 'utf-8'}, function(err,data){
+      if (!err) {
+        let prog = new Program(`${config.file}`);
+        prog.attach("system");
```

Comment:
I should add a way to get a list of all registered watchers so we don't need to touch all these files when we add new first party watchers. The watchers themselves should do feature detection to determine which of their features are supported in the current execution context (browser, headless) and disable things that aren't relevant. That's not blocking this though.